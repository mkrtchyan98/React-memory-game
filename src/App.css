body {
  margin: 0;
  padding: 0;
}
.App {
	display: flex;
	justify-content: center;
	background-color: black;
}
.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.level-btn {
  float:right;
  padding:7px 17px;
  font-weight: 400;
  line-height:normal;
  border-radius:0.1875rem;
  font-size: 1rem;
  background: red;
  translate: transform(-50%,-50%);
  cursor:pointer;
  transition:background 0.2s ease-in;
  margin-top:-2px;
}