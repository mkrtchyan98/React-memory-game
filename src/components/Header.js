import React from 'react';

const Header = ({moves}) => (
  <div className="grid-header-container">
    <div className="justify-left timer"></div>
    <div className="justify-center game-status-text"></div>
    <div className="justify-end">
    	<h1>You won</h1>
		<h2> the expected number of moves needed for a game with n cards converges to ~ 0.8 * n </h2>
	    </div>
  </div>
);

export default Header;