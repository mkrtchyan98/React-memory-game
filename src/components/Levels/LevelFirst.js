import React  from 'react'
import Board from '../board'
import buildCards from '../../buildcards'
import {makeStyles} from '@material-ui/core/styles'

const styles = makeStyles({
 root: {
 	'& div.Board':{
 		width:'100%',
 		height:'100%'
 	}
 }
	})

function LevelFirst() {
	 let  cards = buildCards(4)
	 const classes= styles();
  return (
    <div className={classes.root}>
    <div className="App-header">
 <Board cards={cards}/>
      </div>
    </div>
  )
}
  

export default LevelFirst;