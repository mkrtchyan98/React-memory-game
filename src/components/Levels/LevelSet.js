import React from 'react'

const   LevelSet = (props) => {
  return ( 
    <div>
     <button onClick={() => props.changeLevel("easy")} className="level-btn">Easy</button>
    <button onClick={() => props.changeLevel("medium")} className="level-btn">Medium</button>
    <button onClick={() => props.changeLevel("hard")} className="level-btn">Hard</button>
    </div>
    )
}

export default LevelSet;
