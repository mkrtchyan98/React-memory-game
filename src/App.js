import React, {useState,Suspense} from 'react'
import './App.css';
import LevelSet from './components/Levels/LevelSet';
const LevelFirst = React.lazy(() => import('./components/Levels/LevelFirst'));
const LevelSecond = React.lazy(() => import('./components/Levels/LevelSecond'));
const LevelThird = React.lazy(() => import('./components/Levels/LevelThird'));

function App() {
const [level,setLevel] = useState("easy");
 if(level === "easy"){
  return (
    <div className="App-header">
    <h1>START GAME</h1>
    <LevelSet changeLevel={setLevel} />
          <Suspense fallback={<div>Загрузка...</div>}>
 <LevelFirst />
 </Suspense>
       </div>
 )
}

 if(level === "medium"){
 return (
  <div className="App-header">
    <h1>START GAME</h1>
    <LevelSet changeLevel={setLevel} />
          <Suspense fallback={<div>Загрузка...</div>}>
  <LevelSecond />
  </Suspense>
         </div>

  )
}
 if(level === "hard"){
  return (
   <div className="App-header">
    <h1>START GAME</h1>
    <LevelSet changeLevel={setLevel} />
          <Suspense fallback={<div>Загрузка...</div>}>
  <LevelThird />
  </Suspense>
         </div>

  )

}
}
export default App;