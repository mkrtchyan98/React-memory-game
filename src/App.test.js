import React,{useState} from 'react';
import App from './App';
import Header from './components/Header';
import Board from './components/board'
import LevelFirst from './components/Levels/LevelFirst';
import buildCards from './buildcards'
import LevelSet from './components/Levels/LevelSet';

import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount,shallow } from 'enzyme';

configure({ adapter: new Adapter() });

const deck = buildCards(4);

describe('App',() => {
	test('snapshot renders',() => {
		const component = renderer.create(<App />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	})
})

describe('Header',() => {
	test('snapshot renders',() => {
		const component = renderer.create(<Header />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	})
})

describe('LevelFirst',() => {
	test('snapshot renders',() => {
		const component = shallow(<LevelFirst/>);
		expect(component).toMatchSnapshot();
	});
	it('renders inner Board',() => {
		const wrapper = mount(<LevelFirst />);
		expect(wrapper.find(Board));
	})
})

describe('board',()=>{
	test(' renders',()=> {
 const component = shallow(<Board cards={deck}/>);
		expect(component).toMatchSnapshot();
	})
it('renders props',()=>{
	 const component = mount(<Board cards={deck}/>);
expect(component.prop('cards')).toHaveLength(8);
});
it('renders cards',()=>{
	 const component = mount(<Board cards={deck}/>);
expect(component.find('li')).toHaveLength(8);
});
it('restart moves',()=> {
		 const component = mount(<Board cards={deck}/>);
		 expect(component.find('button').simulate('click'));
		 expect(component.find('div h1').text()).toEqual('Your moves amount:0')

})
})
