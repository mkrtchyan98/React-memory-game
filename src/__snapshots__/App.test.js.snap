// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App-header"
>
  <h1>
    START GAME
  </h1>
  <div>
    <button
      className="level-btn"
      onClick={[Function]}
    >
      Easy
    </button>
    <button
      className="level-btn"
      onClick={[Function]}
    >
      Medium
    </button>
    <button
      className="level-btn"
      onClick={[Function]}
    >
      Hard
    </button>
  </div>
  <div>
    Загрузка...
  </div>
</div>
`;

exports[`Header snapshot renders 1`] = `
<div
  className="grid-header-container"
>
  <div
    className="justify-left timer"
  />
  <div
    className="justify-center game-status-text"
  />
  <div
    className="justify-end"
  >
    <h1>
      You won
    </h1>
    <h2>
       the expected number of moves needed for a game with n cards converges to ~ 0.8 * n 
    </h2>
  </div>
</div>
`;

exports[`LevelFirst snapshot renders 1`] = `ShallowWrapper {}`;

exports[`board  renders 1`] = `ShallowWrapper {}`;
